zerver/lib/integrations.py

```diff
@@ -186,6 +186,40 @@ def url_object(self):
         # type: () -> None
         return
 
+class EmbeddedBotService(Integration):
+    '''
+    This class acts as a registry for bots verified as safe
+    and valid such that these are capable of being deployed on the server.
+    '''
+    IS_ALLOWED = True
+    # This variable informs if this embedded bot service is allowed to run by this
+    # community members. The Organization administrators can turn this off if they
+    # do not want community members to run this bot service.
+
+    def __init__(self, name):
+        # type: (str) -> None
+        super(EmbeddedBotService, self).__init__(
+            name=name,
+            client_name=name,
+            # TODO: Not sure how client_name would be useful here, but it is not None parameter.
+            # setting it to name of the bot for now.
+            categories=['deployment']
+        )
+
+    def is_enabled(self):
+        # type: () -> bool
+        # Not ready to be added to the docs yet.
+        return False
+
+    def is_allowed(self, allow):
+        # type: (bool) -> None
+        self.IS_ALLOWED = allow
```

Comment:
@timabbott IS_ALLOWED will be handled by the organization administration. @showell and I were discussing that the org admins should have the control over what all bots can be used by community members. 
In #organization-settings->Bots , where all the bots are listed, we can have a separate tab, to check what all bots are deployed.

https://chat.zulip.org/#narrow/stream/integrations/subject/embedded.20bots.20future.20tasks/near/251899
This is what I have in mind.
Although on thinking about it again, I can't think of a scenario where a community won't allow the community members to avail a bot.

What are your thoughts?